app/pages/ServiceDiscoveryForm/ServiceDiscoveryForm.tsx

```diff
@@ -4,63 +4,117 @@ import qs from 'qs';
 import ReactGA from 'react-ga';
 
 import { useEligibilitiesForCategory, useSubcategoriesForCategory } from '../../hooks/APIHooks';
+import {
+  CATEGORIES,
+  Step,
+  ServiceCategory,
+} from './constants';
 
-import { CATEGORIES, Step } from './constants';
 import styles from './ServiceDiscoveryForm.module.scss';
 
 interface CategoryRefinement {
   name: string;
   id: number;
 }
 
-interface SelectedResources {
+interface SelectedRefinements {
   [key: number]: boolean;
 }
 
 /** Wrapper component that handles state management, URL parsing, and external API requests. */
 export const ServiceDiscoveryForm = () => {
-  const match = useRouteMatch();
   interface MatchParams {
     categorySlug: string;
   }
 
+  const match = useRouteMatch();
   const { categorySlug } = match.params as MatchParams;
   const category = CATEGORIES.find(c => c.slug === categorySlug);
   if (!category) {
     // Category does not exist; user may have entered the category in the URL bar
-    // or there is an error in our code
     return <Redirect push to={{ pathname: '/' }} />;
   }
 
-  const eligibilities: CategoryRefinement[] = useEligibilitiesForCategory(category.id) || [];
-  const subcategories: CategoryRefinement[] = useSubcategoriesForCategory(category.id) || [];
+  // The activeCategoryId is updated if the user proceeds to a further step that has child
+  // subcategories to be displayed. When it is set, the target subcategory refinements are
+  // fetched and rendered in place of the previous refinements
+  const [activeCategoryId, setActiveCategoryId] = useState<string | null>(category.id);
+
+  // Add the selectedRadioItem to state so that when a user clicks the next button
+  // the selectedRadioItem can be passed to the goToNextStep function
+  const [selectedRadioItem, setSelectedRadioItem] = useState<number | null>(null);
```

Comment:
Not sure if it should go here or somewhere else, but I think a comment somewhere stating that the radio item values are going to be the subcategory ID would be useful, since it's not completely obvious from the code alone.

Reply:
Yes, good suggestion, thanks! I had been considering renaming those vars to `selectedSubcategoryId` but thought that could be a little confusing in relation to the `selectedSubcategories` state var, and that it may not always represent a subcategoryId if the radio component is ever used for eligibilities. 

However, I've gone ahead and updated the state var name to `selectedSubcategoryId` anyhow, since I think it is clearer. When we do a larger refactor of this process, we can consider updating this further